checkov/arm/checks/resource/AzureDefenderOnKubernetes.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+from typing import Any
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.arm.base_resource_check import BaseResourceCheck
+
+
+class AzureDefenderOnKubernetes(BaseResourceCheck):
+    def __init__(self) -> None:
+        name = "Ensure that Azure Defender is set to On for Kubernetes"
+        id = "CKV_AZURE_85"
+        supported_resources = ("Microsoft.Security/pricings",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources,)
+
+    def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
+        return (
+            CheckResult.PASSED
+            if conf.get("name") != "KubernetesService" or str(conf["properties"]["pricingTier"]).lower() == "standard"
```

Comment:
Im not sure that this is the right check - `conf.get("name") != "KubernetesService"`, the name can be everything..

Reply:
I did like the TF check

Reply:
You can remove the `name` from the condition, it is not the same as TF.

Reply:
@ChanochShayner I believe it should only fail if the `"name" : "KubernetesService"` and the pricing tier is not Standard. Otherwise, it will fail for resources that are not K8s based and can't have Defender, despite the name of the policy. It's a very narrow scope, but I believe that's the purpose of the check.